cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretsMetadataStore.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Stores and retrieves metadata for the secrets.
+ *
+ * @param <T> the type of the metadata to be stored.
+ */
+public interface SecretsMetadataStore<T> {
```

Comment:
This is more like `SecretStore` rather than metadata store, as it can be used to store the actual secret.

Reply:
I kept metadata in the name to make sure secrets are not stored in plain text, if user decides to implement `SecretsMetadataStore` in spi implementation. Fixed it and added a note in javadoc.

Reply:
I don't understand. This store API is for the SPI implementation for storing secret, right??

Reply:
To enforce the caller don't call store with plaintext, it would be better to have the store method that takes a `Cipher` class to encrypt the secret.

Reply:
This would mean the store is responsible for encryption/decryption? The store is just a key-value store. For now, I have updated the javadoc.